lib/src/array.h

```diff
@@ -154,7 +154,7 @@ static inline void array__reserve(VoidArray *self, size_t element_size, uint32_t
     } else {
       self->contents = ts_malloc(new_capacity * element_size);
     }
-    self->capacity = new_capacity;
+    self->capacity = (uint32_t)new_capacity;
```

Comment:
Is this cast necessary? It looks like `new_capacity` is already a `uint32_t`

Reply:
This is why we do code review! 😅